ccsrc/lib/cppsim/CMakeLists.txt

```diff
@@ -0,0 +1,305 @@
+set(CPPSIM_INSTALL_SBINDIR "${CMAKE_INSTALL_SBINDIR}")
+set(CPPSIM_INSTALL_SYSCONFDIR "${CMAKE_INSTALL_SYSCONFDIR}")
+set(CPPSIM_INSTALL_INCLUDEDIR "${CMAKE_INSTALL_INCLUDEDIR}/cppsim")
+set(CPPSIM_INSTALL_DATADIR "${CMAKE_INSTALL_DATADIR}/cppsim")
+set(CPPSIM_INSTALL_LIBDIR "${CMAKE_INSTALL_LIBDIR}/cppsim")
+set(CPPSIM_INSTALL_DOCDIR "${CMAKE_INSTALL_DATADIR}/doc/cppsim")
+set(CPPSIM_INSTALL_CMAKEDIR "${CPPSIM_INSTALL_DATADIR}/cmake")
+set(CPPSIM_INSTALL_3RDPARTYDIR "${CPPSIM_INSTALL_LIBDIR}/third_party")
+
+# ==============================================================================
+
+# TODO
+include("${CMAKE_CURRENT_SOURCE_DIR}/../../../cmake/compiler_has_std_filesystem.cmake")
+
+# ==============================================================================
+
+include(FetchContent)
+set(FETCHCONTENT_QUIET OFF)
+
+if(BUILD_TESTING)
+  FetchContent_Declare(
+    googletest
+    GIT_REPOSITORY https://github.com/google/googletest.git
+    GIT_TAG 25cc5777a17820a6339204a3552aa1dd5e428669)
+  FetchContent_Declare(
+    hipThrust
+    GIT_REPOSITORY https://github.com/dmikushin/Thrust.git
+    GIT_TAG 9a12c1259805ed0a3a5fe9bdeb098a872deb936b)
+  FetchContent_MakeAvailable(hipThrust googletest)
+
+  if(NOT DEFINED Eigen3_DIR)
+    FetchContent_Declare(
+      Eigen3
+      GIT_REPOSITORY https://gitlab.com/libeigen/eigen.git
+      GIT_TAG 3147391d946bb4b6c68edd901f2add6ac1f31f8c)
+    FetchContent_MakeAvailable(googletest Eigen3)
+  else()
+    find_package(Eigen3 CONFIG REQUIRED)
+  endif()
+endif()
+
+# ------------------------------------------------------------------------------
+# In the case of digestpp, the repository only consists of a list of files/directories that we need both our build
+# process to find as well as future users of an installed copy of cppsim.
+#
+# We therefore provide a target that we use for our current build and also install its files into a sub-directory of the
+# installation prefix. We then define the digestpp::digestpp target directly within cppsimConfig.cmake
+
+FetchContent_Declare(
+  digestpp
+  GIT_REPOSITORY https://github.com/kerukuro/digestpp.git
+  GIT_TAG 4ec4106677e652a90716ad929d657a622089ef16)
+FetchContent_MakeAvailable(digestpp)
+
+add_library(digestpp::digestpp INTERFACE IMPORTED)
+target_include_directories(digestpp::digestpp INTERFACE $<BUILD_INTERFACE:${digestpp_SOURCE_DIR}>)
+
+install(
+  DIRECTORY ${digestpp_SOURCE_DIR}
+  DESTINATION ${CPPSIM_INSTALL_INCLUDEDIR}/third_party
+  PATTERN docs EXCLUDE
+  PATTERN .git EXCLUDE)
+
+# ------------------------------------------------------------------------------
+
+find_package(res_embed QUIET CONFIG)
+if(NOT res_embed_FOUND)
+  FetchContent_Declare(
+    res_embed
+    GIT_REPOSITORY https://github.com/dmikushin/res_embed.git
+    GIT_TAG 93b5711070086dea53c3b535018ff34e68479242)
+  FetchContent_MakeAvailable(res_embed)
+else()
+  message(STATUS "Found res_embed at ${res_embed_DIR}")
+endif()
```

Comment:
Like for  `Eigen3` these lines are not required since `ResEmbed` is a third-party of MindQuantum.

You can add a short comment saying that one _could want_ to look for `ResEmbed` here if it was not before.